server/files-script/cron-tasks.js

```diff
@@ -38,48 +38,54 @@ const removeDuelsTask = cron.schedule(
 const checkDuelsTask = cron.schedule(
   "* * */3 * * *",
   async function () {
-    const sql = `SELECT duel.du_id, us_login, re_answers, re_last_time \
-               FROM results, duel \
-               WHERE results.du_id = duel.du_id \
-                 AND duel.du_inProgress = 1;`;
+    const sql = `SELECT duel.du_id, du_content, us_login, re_answers, re_last_time \
+                 FROM results, duel \
+                 WHERE results.du_id = duel.du_id \
+                   AND duel.du_inProgress = 1;`;
 
     try {
-      const res = await queryPromise(sql);
-
-      const duelsTime = {};
+      const listOfDuels = {};
 
-      res.forEach((data) => {
-        const lastTime = data.re_last_time;
-        if (!duelsTime[data.du_id]) {
-          duelsTime[data.du_id] = [];
-        }
-        duelsTime[data.du_id].push({
-          user: data.us_login,
-          answers: data.re_answers,
-          time: lastTime === null ? null : new Date(lastTime),
+      const res = await queryPromise(sql);
+      res.forEach((element) => {
+        if (!listOfDuels[element.du_id]) listOfDuels[element.du_id] = [];
+        listOfDuels[element.du_id].push({
+          login: element.us_login,
+          content: element.du_content,
+          answers: element.re_answers,
+          lastTime: element.re_last_time,
         });
       });
 
-      for (const key in duelsTime) {
-        const currentRound = duelsTime[key];
+      await Promise.all(
+        Object.keys(listOfDuels).map(async (duelID) => {
+          const currentDuel = listOfDuels[duelID];
 
-        let indexLastPlayed;
-        if (currentRound[0].time === null || currentRound[1].time === null) {
-          indexLastPlayed = currentRound[0].time ? 0 : 1;
-        } else {
-          indexLastPlayed = currentRound[0].time > currentRound[1].time ? 0 : 1;
-        }
+          let needToPlayID = 0;
```

Comment:
Is it the ID or the index of the player who needs to play ?

Reply:
Yep! Do you want me to change the name of the variable?
I could rename it b

Reply:
I feel like "Var" would be more descriptive